src/components/common/ConfirmAction.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ButtonComponent from './ButtonComponent';
+
+class ConfirmAction extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    const { buttonLoading } = nextProps;
+    if (!buttonLoading) nextProps.toggleModal();
```

Comment:
Please use the braces (`{}`). You can also destructure `toggleModal` as it's part of `nextProps`. I.e.
```
const { buttonLoading, toggleModal } = nextProps;
if (!buttonLoading) {
  return toggleModal();
}